ts/session/objects/ClosedGroup.ts

```diff
@@ -0,0 +1,431 @@
+// This is the (Closed | Medium) Group equivalent to the SessionGroup type.
+
+import { ClosedGroupType, PubKey } from '../types';
+import { UserUtil } from '../../util';
+import { MultiDeviceProtocol } from '../protocols';
+import { onGroupReceived } from '../../receiver/receiver';
+import * as Data from '../../../js/modules/data';
+
+import { createSenderKeysForMembers } from '../medium_group';
+import { GroupUtils, StringUtils } from '../utils';
+import { Constants, Utils } from '..';
+import { ConversationModel } from '../../../js/models/conversations';
+import * as _ from 'lodash';
+
+interface ClosedGroupParams {
+  id: PubKey;
+  type: ClosedGroupType;
+  name: string;
+  admins: Array<PubKey>;
+  members: Array<PubKey>;
+  expireTimer: number;
+}
+
+interface ClosedGroupUpdateParams {
+  // Closed groups do not currently support avatars
+  name: string;
+  avatar?: string;
+  recipients: Array<PubKey>;
+  members: Array<PubKey>;
+  isMediumGroup: boolean;
+  options: any;
+}
+
+export class ClosedGroup {
+  public readonly id: PubKey;
+  public readonly type: ClosedGroupType;
+  private _name: string;
+  private _admins: Array<PubKey>;
+  private _members: Array<PubKey>;
+  private _expireTimer: number;
+
+  constructor(params: ClosedGroupParams) {
+    // Setters and getters are used to prevent altering the values of name, admins, etc without a group update.
```

Comment:
I don't know if that's how it should work. I don't think we should send N network requests whenever we change a field in this object. Something as expensive as this should be explicit.